tests/file_yaml.rs

```diff
@@ -35,6 +37,35 @@ fn make() -> Config {
         .unwrap()
 }
 
+#[test]
+fn test_keys_with_periods_deserialize_serde_yaml() {
+    let map = "192.168.1.1: a string value";
+
+    let c: HashMap<String, String> = serde_yaml::from_str(map).unwrap();
+
+    assert_eq!(c.get("192.168.1.1").unwrap(), "a string value");
+}
+
+#[test]
+fn test_keys_with_periods_deserialize_yaml_rust() {
+    use yaml_rust::YamlLoader;
+
+    let map = "192.168.1.1: a string value";
+
+    let c = YamlLoader::load_from_str(map).unwrap().first().unwrap().clone();
+
+    assert_eq!(c["192.168.1.1"].as_str().unwrap(), "a string value");
+}
+
+#[test]
+fn test_keys_with_periods_deserialize() {
+    let c = make();
+
+    let s: Settings = c.try_deserialize().unwrap();
```

Comment:
This test fails here because the `tests/Settings` files are not updated.

Reply:
@matthiasbeyer It was updated, but due to #417 it fails.  This is the bug reproduced.

https://github.com/Halkcyon/config-rs/blob/cdb945f8a2b2fd7223d518eec874e28e38e491fa/tests/Settings.yaml#L18

Reply:
Ah, I guess I was not fully awake when reviewing this. Sorry about that.